IoT-Sample/IoTAmplifySample/IoTAmplifySample/Auth/AuthViewModel.swift

```diff
@@ -0,0 +1,92 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Combine
+import Foundation
+
+final class AuthViewModel: ObservableObject {
```

Comment:
Why not add `@MainActor` for the entire class.  That allows to avoid having to use `MainActor.run()` in all methods and it is safer for future changes as well (dev can not forget to use `MainActor.run()` in future methods)

Reply:
Good point - I will wrap this in a `@MainActor`!